deso/block.go

```diff
@@ -487,6 +491,66 @@ func (node *Node) getAcceptNFTOps(txn *lib.MsgDeSoTxn, utxoOpsForTxn []*lib.Utxo
 	return operations
 }
 
+func (node *Node) getBuyNowNFTBidOps(txn *lib.MsgDeSoTxn, utxoOpsForTxn []*lib.UtxoOperation, numOps int) []*types.Operation {
+	if txn.TxnMeta.GetTxnType() != lib.TxnTypeNFTBid {
+		return nil
+	}
+
+	// Extract the NFTBid op
+	var nftBidOp *lib.UtxoOperation
+	for _, utxoOp := range utxoOpsForTxn {
+		if utxoOp.Type == lib.OperationTypeNFTBid {
+			nftBidOp = utxoOp
+			break
+		}
+	}
+	if nftBidOp == nil {
+		fmt.Printf("Error: Missing UtxoOperation for NFTBid txn: %v\n", txn.Hash())
+		return nil
+	}
+
+	// We only care about NFT bids that generate creator royalties. This only occurs for NFT bids on Buy Now NFTs.
+	// Only NFT bids on Buy Now NFTs will have NFTBidcreatorRoyaltyNanos > 0. Even in the case that an NFT is a buy now,
+	// it is possible that there are no royalties to the creator coin generated. In this situation, it is okay to exit
+	// early since there are not output to add.
+	if nftBidOp.NFTBidCreatorRoyaltyNanos == 0 {
+		return nil
+	}
+
+	var operations []*types.Operation
+
+	royaltyAccount := &types.AccountIdentifier{
+		Address: lib.PkToString(nftBidOp.NFTBidCreatorPublicKey, node.Params),
+		SubAccount: &types.SubAccountIdentifier{
+			Address: CreatorCoin,
+		},
+	}
+
+	// Add an output representing the creator coin royalty only if there
+	// are enough creator coins in circulation
+	//
+	// TODO: This if statement is needed temporarily to fix a bug whereby
+	// NFTBidCreatorRoyaltyNanos is non-zero even when the royalty given
+	// was zero due to this check in consensus.
+	if nftBidOp.PrevCoinEntry.CoinsInCirculationNanos >= node.Params.CreatorCoinAutoSellThresholdNanos {
```

Comment:
@diamondhands0 did you write this if statement for `TxnTypeAcceptNFTBid`? Does it still apply to Buy Now NFTs?